docs/howto/operate/ci-cd/split-repo2docker.md

```diff
@@ -0,0 +1,77 @@
+# Split up an image for use with r2d-action
+
+Sometimes we have user images defined in a repo, and we want to extract
+it to a standalone repo so it can be used with the [repo2docker action](https://github.com/jupyterhub/repo2docker-action).
+But we want to retain full history of the image still, so we can look
+back and see why things are the way they are, as well as credit the people
+who contributed over time. This page documents the git-fu required to
+make this happen.
+
+1. Clone the base repository you are extracting the image from.
+   We will be performing destructive operations on this clone,
+   so it has to be a fresh clone.
+
+    ```bash
+    git clone https://github.com/utoronto-2i2c/jupyterhub-deploy utoronto-image --origin source
+    ```
+
+    We name the directory `utoronto-image` as that would be the name of the
+    repo with just the user image. We ask git to name the remote it creates be
+    called `source`, as we will create a new GitHub repo later that will be `origin`.
+
+2. Make sure [git-filter-repo](https://github.com/newren/git-filter-repo)
+   is installed. It should be available in [brew or other package managgers]
+   (https://github.com/newren/git-filter-repo/blob/main/INSTALL.md)
+
+3. Run `git-filter-repo` to remove everything from the repo except the
+   image directory.
+
+   ```bash
+    git filter-repo --subdirectory-filter deployments/utoronto/image --force
+    ```
+
+    The repo root directory now contains the contents of `deployments/utoronto/image`,
+    as well as full git history for any commits that touched it! This way,
+    we do not lose history or attribution.
+
+4. Create a new repository in GitHub based off [this template](https://github.com/2i2c-org/hub-user-image-template).
+   The template makes it much easier to setup repo2docker-action on it.
+
+5. Add the new repo as a remote you can push commits to / pull commits from.
+   This will be the primary location of this repo now, so let's call it `origin`.
+
+   ```bash
+   git remote add origin git@github.com:2i2c-org/utoronto-image.git
```

Comment:
```suggestion
   git remote add origin git@github.com:2i2c-org/<hub-image>.git
```

Reply:
Does it have to be the 2i2c-org? I think this command could be:

```
git remote add origin git@github.com:<your-github-username-or-org>/<hub-image>.git
```

Reply:
I've done what @sgibson91 suggested here.